flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGate.java

```diff
@@ -561,6 +567,7 @@ public void requestPartitions() throws IOException, InterruptedException {
 		}
 
 		final Buffer buffer = result.get().buffer();
+		numBytesIn.inc(buffer.getSizeUnsafe());
```

Comment:
here we now count more stuff, but aren't counting less anywhere else. Why is that?

Reply:
This should be just yet another new total counter for all tasks (local + remote channels). Old counters with local and remote separation should stay as internal specific metrics of network environment. At the end of the day, total = local + remote and used directly without summing of specific counters outside of network environment in IOMetrics.

Reply:
Later we plan to move the total counter completely out of network environment to count it always in gate wrapper to make it non-shuffle specific: https://issues.apache.org/jira/browse/FLINK-12201

Reply:
exactly, thanks for replies. @azagrebin

Reply:
We add a new total metric used for all the existing handlers which keep the same behavior as before, then the existing local/remote metrics inside `NetworkEnvironment` could also be visible for outside users.